src/app/exercise/editors/editors.component.ts

```diff
@@ -20,6 +20,29 @@ export class EditorsComponent {
     this.debug = true;
   }
 
+  ngOnInit() {
+    this.files.map( file => {
+      file.collapsed = !!file.readonly;
+    });
+  }
+
+  closeTab(file) {
+    file.collapsed = true;
+
+    if (file === this.currentFile) {
+      this.showFile(
+        this.visibleFiles
```

Comment:
@kirjs Actually while testing realized that `this.visibleFiles[this.visibleFiles.lastIndexOf(file)]` is not going to get the right file. I think the name of the array `this.visibleFiles` became  misleading after the updates we made to the editor. It can make you think that 'visible' are the ones that are opened, when in fact it's the files that can be viewed and/or edited, the files in the file tree. Should I change the `visibleFiles` to `fileTree` or any other name that you think will describe the array's content better?

Since I need the closest file to the active file that is being closed, I need to look up a file in the ActiveArray that is also currently in open state. I agree that creating new copy of the array every time is not best BUT it's not being saved to any variable, thus it shouldn't add garbage to the local memory (correct me if I'm wrong). Sooo I'm thinking to leave it as it is for now.. What you say? :D